src/utils/upload.js

```diff
@@ -82,4 +84,39 @@ export default class Upload {
     this.offset = (await this.$rest.get(`file/offset?uploadId=${this.upload._id}`)).data.offset;
     return this._sendChunks();
   }
+
+  /**
+   * This callback is called before the upload is started. This callback is asynchronous.
+   * If it returns a Promise, the caller will await its resolution before continuing.
+   * This callback is passed the index of this file in the overall list of files being uploaded,
+   * which is mostly done to allow asynchronous hooks to be performed only before the first file or
+   * after the last file, which is a common use case.
+   * @param current {Number} The index of this file in the list.
+   * @param total {Number} The total number of files being uploaded.
+   */
+  // eslint-disable-next-line class-methods-use-this, no-unused-vars
+  beforeUpload({ current, total }) {}
+
+  /**
+   * This callback is called after the upload is completed. This callback is asynchronous.
+   * If it returns a Promise, the caller will await its resolution before continuing.
+   * This callback is passed the index of this file in the overall list of files being uploaded,
+   * which is mostly done to allow asynchronous hooks to be performed only before the first file or
+   * after the last file, which is a common use case.
```

Comment:
Move any relevant info from 103 line into the @param line and remove?

Remove 104, 105, and also from 91,92.  I think it's fairly clear how to use these hooks.  

Or maybe put it into the header docstring for this file?